x-pack/solutions/security/plugins/security_solution/public/asset_inventory/components/asset_inventory_data_table.tsx

```diff
@@ -167,13 +150,14 @@ export const AssetInventoryDataTable = ({ state }: AssetInventoryDataTableProps)
     onFlyoutClose: () => setExpandedDoc(undefined),
   });
 
-  const onExpandDocClick = (record?: DataTableRecord | undefined) => {
-    if (record) {
-      const { entity, source } = getEntity(record);
-      setExpandedDoc(record); // Table is expecting the same doc ref to highlight the selected row
+  const openTableFlyout = (doc?: DataTableRecord | undefined) => {
+    if (doc && doc.raw._source) {
+      const source = doc.raw._source as GenericEntityRecord;
```

Comment:
asserting since `DataTableRecord` does not accept a generic. Could edit it or use a custom type by omitting some fields, but that feels out of scope